### PR TITLE
some code refactoring

### PR DESCRIPTION
**Pull request by guolinke**

### CODE REVIEW

File: src/treelearner/serial_tree_learner.cpp [hunk 1]
```diff
@@ -407,15 +409,18 @@ void SerialTreeLearner::FindBestSplitsFromHistograms(const std::vector<int8_t>&
     larger_node_used_features = GetUsedFeatures(false);
   }
   OMP_INIT_EX();
-  // find splits
-  #pragma omp parallel for schedule(static)
+// find splits
+#pragma omp parallel for schedule(guided)
```
Comment (guolinke) [inline src/treelearner/serial_tree_learner.cpp]:
```suggestion
#pragma omp parallel for schedule(static)
```

File: src/treelearner/serial_tree_learner.h [hunk 2]
```diff
@@ -63,10 +63,13 @@ class SerialTreeLearner: public TreeLearner {
...
-  void AddPredictionToScore(const Tree* tree, double* out_score) const override {
-    if (tree->num_leaves() <= 1) { return; }
+  void AddPredictionToScore(const Tree* tree,
+                            double* out_score) const override {
+    if (tree->num_leaves() <= 1) {
+      return;
+    }
     CHECK(tree->num_leaves() <= data_partition_->num_leaves());
-    #pragma omp parallel for schedule(static)
+#pragma omp parallel for schedule(guided)
```
Comment (guolinke) [inline src/treelearner/serial_tree_learner.h]:
```suggestion
#pragma omp parallel for schedule(static, 1)
```

File: src/treelearner/data_partition.hpp [hunk 3]
```diff
@@ -114,18 +115,19 @@ class DataPartition {
...
     // get leaf boundary
     const data_size_t begin = leaf_begin_[leaf];
     const data_size_t cnt = leaf_count_[leaf];
 
-    const int nblock = std::min(num_threads_, (cnt + min_inner_size - 1) / min_inner_size);
-    data_size_t inner_size = SIZE_ALIGNED((cnt + nblock - 1) / nblock);
+    int nblock = 1;
+    data_size_t inner_size = cnt;
+    Threading::BlockInfo<data_size_t>(num_threads_, cnt, 512, &nblock,
+                                        &inner_size);
```
Comment (guolinke) [inline src/treelearner/data_partition.hpp]:
```suggestion
                                      &inner_size);
```

File: src/io/dataset.cpp [hunk 4]
```diff
@@ -1158,11 +1194,10 @@ void Dataset::ConstructHistogramsMultiVal(
...
       (num_bin + kAlignedSize - 1) / kAlignedSize * kAlignedSize;
-  const int min_data_block_size = 1024;
-  const int n_data_block = std::min(
-      num_threads, (num_data + min_data_block_size - 1) / min_data_block_size);
-  const int data_block_size = (num_data + n_data_block - 1) / n_data_block;
-
+  int n_data_block = 1;
+  int data_block_size = num_data;
+  Threading::BlockInfo<data_size_t>(num_threads, num_data, 1024,
+                                      &n_data_block, &data_block_size);
```
Comment (guolinke) [inline src/io/dataset.cpp]:
```suggestion
                                    &n_data_block, &data_block_size);
```

File: src/io/dataset.cpp [hunk 5]
```diff
@@ -1198,14 +1235,16 @@ void Dataset::ConstructHistogramsMultiVal(
...
 
   global_timer.Start("Dataset::sparse_bin_histogram_merge");
-  const int min_bin_block_size = 512;
-  const int n_bin_block = std::min(
-      num_threads, (num_bin + min_bin_block_size - 1) / min_bin_block_size);
-  const int bin_block_size = (num_bin + n_bin_block - 1) / n_bin_block;
+  int n_bin_block = 1;
+  int bin_block_size = num_bin;
+  Threading::BlockInfo<data_size_t>(num_threads, num_bin, 512, &n_bin_block,
+                                      &bin_block_size);
```
Comment (guolinke) [inline src/io/dataset.cpp]:
```suggestion
                                    &bin_block_size);
```

**Comment by guolinke** [comment]
for the code formatting, I switch to the the "Google" style in visual studio:

image

**Comment by StrikerRUS** [comment]
@guolinke 
> for the code formatting, I switch to the "Google" style in visual studio:

Nice! I hope we will be able to resolve the issue #1990 soon.

File: src/treelearner/feature_histogram.hpp [hunk 6]
```diff
@@ -565,26 +565,29 @@ class FeatureHistogram {
...
-          // better split point
-          if (current_gain > best_gain) {
-            best_left_count = left_count;
-            best_sum_left_gradient = sum_left_gradient;
-            best_sum_left_hessian = sum_left_hessian;
-            // left is <= threshold, right is > threshold.  so this is t-1
-            best_threshold = static_cast<uint32_t>(t - 1 + offset);
-            best_gain = current_gain;
+        if (is_rand) {
+          if (t + offset != rand_threshold) {
```
Comment (StrikerRUS) [inline src/treelearner/feature_histogram.hpp]: @guolinke Wasn't `-1` lost here?

Comment (StrikerRUS) [reply src/treelearner/feature_histogram.hpp]: @guolinke

Comment (guolinke) [reply src/treelearner/feature_histogram.hpp]: Oh, you are right!

**Pull request merged**